src/lib/giantsFoundry.ts

```diff
@@ -0,0 +1,75 @@
+import { randInt } from 'e';
+
+import { logError } from './util/logError';
+
+// Difference between this and ItemBank, is ItemBank is expected to have only integers as strings
+export interface GiantsFoundryBank {
+	[key: string]: number;
+}
+export const tipMoulds: string[] = [
+	'Saw Tip',
+	'Gladius Point',
+	"Serpent's Fang",
+	"Medusa's Head",
+	'Chopper Tip',
+	'People Poker Point',
+	'Corrupted Point',
+	"Defender's Tip",
+	'Serrated Tip',
+	'Needle Point',
+	'The Point!'
+];
+
+export const bladeMoulds: string[] = [
+	'Gladius Edge',
+	'Stiletto Blade',
+	'Medusa Blade',
+	'Fish Blade',
+	"Defender's Edge",
+	'Saw Blade',
+	'Flamberge Blade',
+	'Serpent Blade',
+	'Claymore Blade',
+	'Fleur de Blade',
+	'Choppa!'
+];
+
+export const forteMoulds: string[] = [
+	'Serrated Forte',
+	'Serpent Ricasso',
+	'Medusa Ricasso',
+	'Disarming Forte',
+	'Gladius Ricasso',
+	'Chopper Forte',
+	'Stiletto Forte',
+	"Defender's Base",
+	"Juggernaut's Forte",
+	'Chopper Forte +1',
+	'Spiker!'
+];
+
+export const TOTAL_GIANT_WEAPONS = tipMoulds.length * bladeMoulds.length * forteMoulds.length;
+
+// weaponID stored as hex 0a0403 => 10, 4, 3
+export function decodeGiantWeapons(weaponID: string) {
+	const decimal = parseInt(weaponID, 16);
+	const [tipMould, bladeMould, forteMould] = [decimal >> 16, (decimal >> 8) & 255, decimal & 255];
+	return [tipMould, bladeMould, forteMould];
+}
```

Comment:
I mostly wanted to learn bitwise and hex conversion to improve my coding skill and with Cyrs help understanding it I really like the way it's stored, it makes the length of the keys same size always rather than being of different lengths with like ''11-0-2'' and ''10-10-10''.

I can still change back to that way if you want, I tested it out and it works great and is fast.

Reply:
I'd rather it changed back. It's good you learned and that it makes things the same length, but besides that, its little benefit for a much more complex way of doing it.

Reply:
changed back